dashboard/modules/job/tests/test_job_agent.py

```diff
@@ -518,5 +522,57 @@ def test_agent_logs_not_streamed_to_drivers():
     assert "(raylet)" not in err_str
 
 
+@pytest.mark.asyncio
+async def test_non_default_dashboard_agent_http_port(tmp_path):
+    """Test that we can connect to the dashboard agent with a non-default
+    http port.
+    """
+    import subprocess
+
+    cmd = (
+        "ray start --head --block "
+        f"--dashboard-agent-listen-port {get_current_unused_port()}"
+    )
+    proc = subprocess.Popen(cmd.split(" "))
+
+    async def verify():
+        address_info = ray.init("auto", ignore_reinit_error=True).address_info
+
+        ip, _ = address_info["webui_url"].split(":")
+        dashboard_agent_listen_port = address_info["dashboard_agent_listen_port"]
+        agent_address = f"{ip}:{dashboard_agent_listen_port}"
+        print(agent_address)
+
+        agent_client = JobAgentSubmissionClient(format_web_url(agent_address))
+
+        assert wait_until_server_available(agent_address)
+
+        # Submit a job through the agent.
+        runtime_env = RuntimeEnv().to_dict()
+        request = validate_request_type(
+            {
+                "runtime_env": runtime_env,
+                "entrypoint": "echo hello",
+            },
+            JobSubmitRequest,
+        )
+
+        submit_result = await agent_client.submit_job_internal(request)
+
+        job_id = submit_result.submission_id
+        print(job_id)
+        resp = await agent_client.get_job_logs_internal(job_id)
+        assert "hello" in resp.logs, resp.logs
+
+        return True
+
+    try:
+        await async_wait_for_condition_async_predicate(verify, retry_interval_ms=2000)
```

Comment:
Which part do we really intend to retry here? Does it make sense to do the setup once, and only retry the part that checks if the logs have appeared yet?